app/controllers/components/course/materials_component.rb

```diff
@@ -0,0 +1,15 @@
+class Course::MaterialsComponent < SimpleDelegator
+  include Course::ControllerComponentHost::Component
+
+  def sidebar_items
+    [
+      {
+        key: :materials,
+        title: I18n.t('course.material.sidebar_title'),
+        weight: 4,
+        path: course_material_folder_path(current_course, current_course.root_folder),
```

Comment:
Google around. Many answers.

In short: migration can be run after the model changes. Then your migration is broken.